marklogic-client-api/src/main/java/com/marklogic/client/datamovement/impl/WriteBatcherImpl.java

```diff
@@ -1339,16 +1341,13 @@ public boolean awaitCompletion(long timeout, TimeUnit unit) throws InterruptedEx
 
 @Override
 public WriteBatcher withDefaultMetadata(DocumentMetadataHandle handle) {
-    this.docMetadataHandle = handle;
+    this.defaultMetadata = handle;
     return this;
 }
 
 @Override
-public void addAll(Stream<? extends DocumentWriteOperation> operations, JacksonCSVSplitter splitter) {
+public void addAll(Stream<? extends DocumentWriteOperation> operations) {
     Iterator<? extends DocumentWriteOperation> docWriteItr = operations.iterator();
-    while(docWriteItr.hasNext()) {
-      add(docWriteItr.next());
-      splitter.incrementCount();
-    }
+    docWriteItr.forEachRemaining(this::add);

```

Comment:
Would it work to drop the iterator and just execute

`    operations.forEach(this::add);`